modules/meta.analysis/R/run.meta.analysis.R

```diff
@@ -6,129 +6,136 @@
   logger.info(" Running meta.analysis for PFT:", pft$name)
   logger.info("-------------------------------------------------------------------")
   
   ## Load trait data for PFT
-  load(file.path(pft$outdir, 'trait.data.Rdata'))
-  load(file.path(pft$outdir, 'prior.distns.Rdata'))
-
-  if(length(trait.data) == 0) {
+  load(file.path(pft$outdir, "trait.data.Rdata"))
+  load(file.path(pft$outdir, "prior.distns.Rdata"))
+  
+  if (length(trait.data) == 0) {
     logger.info("no trait data for PFT", pft$name, "\n so no meta-analysis will be performed")
     return(NA)
   }
-
+  
   # create path where to store files
   pathname <- file.path(dbfiles, "posterior", pft$posteriorid)
   dir.create(pathname, showWarnings = FALSE, recursive = TRUE)
-
+  
   ## Convert data to format expected by pecan.ma
   jagged.data <- lapply(trait.data, jagify)
-
+  
   ## Check that data is consistent with prior
-  for(trait in names(jagged.data)){
-    data.median    <- median(jagged.data[[trait]]$Y)
-    prior          <- prior.distns[trait, ]
-    p.data         <- p.point.in.prior(point = data.median, prior = prior)
-    if(p.data <= 0.9995 & p.data >= 0.0005){
+  for (trait in names(jagged.data)) {
+    data.median <- median(jagged.data[[trait]]$Y)
+    prior       <- prior.distns[trait, ]
+    p.data      <- p.point.in.prior(point = data.median, prior = prior)
+    
+    if (p.data <= 0.9995 & p.data >= 5e-04) {
       if (p.data <= 0.975 & p.data >= 0.025) {
         logger.info("OK! ", trait, " data and prior are consistent:")
       } else {
         logger.warn("CHECK THIS: ", trait, " data and prior are inconsistent:")
       }
-    } else if (p.data > 0.9995 | p.data < 0.0005) {
-      logger.debug("NOT OK! ", trait," data and prior are probably not the same:")
+    } else if (p.data > 0.9995 | p.data < 5e-04) {
+      logger.debug("NOT OK! ", trait, " data and prior are probably not the same:")
       return(NA)
     }
     logger.info(trait, "P[X<x] =", p.data)
   }
   
   ## Average trait data
-  trait.average <- sapply(jagged.data,
-                          function(x){mean(x$Y, na.rm = TRUE)})
+  trait.average <- sapply(jagged.data, function(x) mean(x$Y, na.rm = TRUE) )
   
   ## Set gamma distribution prior
-  prior.variances = as.data.frame(rep(1, nrow(prior.distns)))
-  row.names(prior.variances) = row.names(prior.distns)
-  prior.variances[names(trait.average), ] = 0.001 * trait.average^2 
-  prior.variances["seedling_mortality", 1] = 1.0
-  taupriors <- list(tauA = 0.01,
-                    tauB = apply(prior.variances, 1, function(x) min(0.01, x)))
+  prior.variances <- as.data.frame(rep(1, nrow(prior.distns)))
+  row.names(prior.variances) <- row.names(prior.distns)
+  prior.variances[names(trait.average), ] <- 0.001 * trait.average ^ 2
+  prior.variances["seedling_mortality", 1] <- 1
+  taupriors <- list(tauA = 0.01, tauB = apply(prior.variances, 1, function(x) min(0.01, x)))
```

Comment:
@ashiklom this is not only magical, it is a crude hack - if the means are very small the priors are too strong. Not sure what the 'best' approach would be, but rescaling the data (and unscaling the results) is one option.